### PR TITLE
Use define macro in setup.py

### PR DESCRIPTION
**Pull request by okuta**
This PR is related to #1117.

### CODE REVIEW

File: install/build.py [hunk 1]
```diff
@@ -367,50 +367,39 @@ def check_nvtx(compiler, settings):
...
-        fname = os.path.join(temp_dir, 'a.cpp')
-        with open(fname, 'w') as f:
-            f.write(source)
-
-        objects = compiler.compile([fname], output_dir=temp_dir,
-                                   include_dirs=include_dirs)
-
-        try:
-            postargs = ['/MANIFEST'] if sys.platform == 'win32' else []
-            compiler.link_executable(objects,
```
Comment (unnonouno) [inline install/build.py]: `build_and_run` calls `link_executable`, but `build_shlib ` calls `link_shared_lib `. You cannot change it.

Comment (okuta) [reply install/build.py]: OK, I will fix it.

**Comment by okuta** [comment]
I fixed.

**Comment by unnonouno** [comment]
LGTM

**Comment by unnonouno** [comment]
Do we need to backport it?

**Comment by okuta** [comment]
No. I removed to-be-backported label.

**Pull request merged**